### PR TITLE
Moving to lezer v0.12.0 through codemirror.next v0.14.0

### PR DESCRIPTION
**Pull request by Nexucis**
This is a deep review of how we should iterate over the tree.

Note: as a side effect, it looks like the resolution of the tree is much more accurate and make more sense according to different weird case we had to patch with ugly code. ( like the autocompletion of the `MatchOp` )

And this is really cool we have many tests. Otherwise it would have been really hard to find what was broken after making this move.

### CODE REVIEW

**Comment by Nexucis** [comment]
and perhaps there is a way to improve it by using a bit more the feature from lezer, but I don't have better idea for the moment. At least if someone want to do it, it's now possible.

File: src/lang-promql/parser/parser.ts [hunk 1]
```diff
@@ -83,33 +83,28 @@ export class Parser {
...
     // So to be able to iterate other the node relative to the promql node, we have to get the first child at the beginning
-    this.checkAST(this.tree.firstChild);
+    this.checkAST(this.tree.topNode.firstChild);
     this.diagnoseAllErrorNodes();
   }
 
   private diagnoseAllErrorNodes() {
-    this.tree.iterate({
-      enter: (type, start, end) => {
-        // usually there is an error node at the end of the expression when user is typing
```
Comment (juliusv) [inline src/lang-promql/parser/parser.ts:93]: Could you keep this comment? Makes it much easier to understand the purpose :)

Comment (Nexucis) [reply src/lang-promql/parser/parser.ts:93]: done

**Comment by juliusv** [comment]
:+1: Awesome :)

**Comment by juliusv** [comment]
(I also played around a bit with it and couldn't find any breakage yet)

**Pull request merged**